maze_transformer/training/dataset.py

```diff
@@ -115,17 +158,305 @@ def from_sequences(cls, data: list[ATensor[("tokens")]]) -> "IndexedArray":
         return cls(arr=arr, indices=indices)
 
 
+class SaveFormats(enum.Enum):
+    OBJECTS: str = "objects"
+    TOKENS: str = "tokens"
+    ARRAY: str = "array"
+
+
 class GPTDataset(Dataset):
```

Comment:
Lots of great stuff in here, but as far as I can tell this class has no unit tests at all? I'd definitely like to see those added before merging.